openstef/pipeline/train_create_forecast_backtest.py

```diff
@@ -32,150 +35,51 @@ def train_model_and_forecast_back_test(
-            )
 
-            models_entire.append(model)
-            valid_data_entire.append(validation_data)
-            train_data_entire.append(train_data)
-            test_data_entire.append(test_data)
+    data_with_features = train_model.train_pipeline_compute_features(
+        input_data=input_data, pj=pj, modelspecs=modelspecs, horizons=training_horizons
+    )
 
-            forecast = forecast.append(forecast_fold).sort_index()
-    else:
-        (
-            forecast,
-            model,
-            train_data,
-            validation_data,
-            test_data,
-        ) = train_model_and_forecast_test_core(
-            pj,
-            modelspecs,
-            input_data,
-            training_horizons=training_horizons,
+    (
+        models_folds,
+        forecast_folds,
+        train_data_folds,
+        validation_data_folds,
+        test_data_folds,
+    ) = zip(
+        *(
+            train_model_and_forecast_test_core(
+                pj, modelspecs, train_data, validation_data, test_data
+            )
+            + (train_data, validation_data, test_data)
+            for train_data, validation_data, test_data in backtest_split_func(
+                data_with_features, n_folds, **backtest_split_args
+            )
         )
-
-        models_entire.append(model)
-        valid_data_entire.append(validation_data)
-        train_data_entire.append(train_data)
-        test_data_entire.append(test_data)
+    )
 
     return (
-        forecast,
-        models_entire,
-        train_data_entire,
-        valid_data_entire,
-        test_data_entire,
+        pd.concat(forecast_folds, axis=0).sort_index(),
+        list(models_folds),
+        list(train_data_folds),
+        list(validation_data_folds),
+        list(test_data_folds),
     )
 
 
 def train_model_and_forecast_test_core(
-    pj: PredictionJobDataClass,
-    modelspecs: ModelSpecificationDataClass,
-    input_data: pd.DataFrame,
-    training_horizons: List[float] = None,
-    test_data: pd.DataFrame = pd.DataFrame(),
-    test_fraction: float = 0.15,
-) -> Tuple[pd.DataFrame, OpenstfRegressor]:
-    """Core part of the backtest pipeline, in order to create a model and forecast from input data
-
-    Args:
-        pj (PredictionJobDataClass): Prediction job.
-        modelspecs (ModelSpecificationDataClass): Dataclass containing model specifications
-        input_data (pd.DataFrame): Input data
-        training_horizons (list): horizons to train on in hours.
-            These horizons are also used to make predictions (one for every horizon)
-
-    Returns:
-        forecast (pandas.DataFrame)
-
-    """
-    model, report, train_data, validation_data, test_data = train_pipeline_common(
-        pj,
-        modelspecs,
-        input_data,
-        training_horizons,
-        test_fraction=test_fraction,
-        backtest=True,
-        test_data_predefined=test_data,
+    pj, modelspecs, train_data, validation_data, test_data
```

Comment:
Also here, please add some documentation (can be short) and type hints so it is clear how to use this function.